src/server.js

```diff
@@ -0,0 +1,319 @@
+      ];
+
+      // Merge the chromium default and balena default flags
+      flags = flags.concat(balenaFlags);
+    }
+
+    if (enableGpu != '1')
+    {
+      console.log("Disabling GPU")
+      flags.push('--disable-gpu')
+    }
+    else
+    {
+      console.log("Enabling GPU")
+    }
+
+    if (PERSISTENT_DATA == '1')
+    {
+      flags.push('--user-data-dir=/data')
+    }
+
+    var startingUrl = url;
+    if (kioskMode == '1')
+    {
+      console.log("Enabling KIOSK mode")
+      startingUrl = '--app=' + url
+    }
+    else
+    {
+      console.log("Disabling KIOSK mode")
+    }
+
+    console.log("Starting Chromium with the following flags: " + flags)
+    console.log("Displaying URL " + startingUrl)
+
+    chromeLauncher.launch({
+      startingUrl: startingUrl,
+      ignoreDefaultFlags: true,
+      chromeFlags: flags,
+      port: REMOTE_DEBUG_PORT,
+      userDataDir: '/tmp/balena'
+    }).then(chrome => {
+      console.log(`Chromium debugging port running on port: ${chrome.port}`);
+      currentUrl = url;
+    });
+  });
+}
+
+// Get's the chrome-launcher default flags, minus the extensions and audio muting flags.
+async function SetDefaultFlags() {
+  DEFAULT_FLAGS =  await chromeLauncher.Launcher.defaultFlags().filter(flag => flag !== '--disable-extensions' && flag !== '--mute-audio');
+}
+
+async function main(){
+  await SetDefaultFlags();
+  var url = await getUrlToDisplayAsync();
+  launchChromium(url);
+}
+
+main().catch("Main method error: " + console.log);
+
+// Start the API
+const app = express();
+
+const errorHandler = (err, req, res, next) => {
+  res.status(500);
+  res.render('API error: ', {
+    error: err
+  });
+};
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+app.use(function(req, res, next) {
+  res.header("Access-Control-Allow-Origin", "*");
+  res.header('Access-Control-Allow-Methods', 'GET,PUT,POST,DELETE');
+  res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+  next();
+});
+app.use(errorHandler);
+
+// ping endpoint
+app.get('/ping', (req, res) => {
+    
+    return res.status(200).send('ok');
+});
+
+// url set endpoint
+app.post('/url', (req, res) => {
+  if (!req.body.url) {
+    return res.status(400).send('Bad request: missing URL in the body element');
+  }
+
+  var url = req.body.url;
+
+  // prepend http prefix if necessary for kiosk mode to work
+  if (!/^https?:\/\//i.test(url)) {
+    url = 'http://' + url;
+  }
+
+  if (req.body.kiosk) {
+    kioskMode = req.body.kiosk;
+  }
+
+  if (req.body.gpu) {
+    enableGpu = req.body.gpu;
+  }
+
+  launchChromium(url);
+  return res.status(200).send('ok');
+});
+
+// url get endpoint
+app.get('/url', (req, res) => {
+    
+  return res.status(200).send(currentUrl);
+});
+
+// refresh endpoint
+app.post('/refresh', (req, res) => {
+ 
+  launchChromium(currentUrl);
+  return res.status(200).send('ok');
+});
```

Comment:
Here is an example of what I mentioned previously. 

`return res.status(200).send('ok');` won't wait for `launchChromium` internal promises to be resolved, so it could fail after you send the `200`.